### ISSUE TITLE
Method calls with enumeration types don't work as expected

### ISSUE THREAD

**Issue by marcodisarno** [node-opcua/node-opcua#1086]
**Current behavior**

When adding enumeration types as input arguments for method calls, they always fail:

**Describe the bug**

We get an BadInvalidArgument as a status code and the bound callback is not executed.

Reason seems to be that the client sends back an integer type, while NodeOPCUA server expects the enum datatype. I think, however, the client's behaviour is okay, since numerical values could easily be mapped to the enum type.

**Step by step instruction to reproduce**

1. Start an NodeOPCUA server
2. Connect a client (UAExpert, FreeOPCUA)
3. Go to Server -> RequestServerStateChange (or any other Method using enumeration type input parameters)
4. Call it

**Expected behavior**

Bound method should be called with chosen input parameter.

**Screenshots**

This is from UAExpert:
![image](https://user-images.githubusercontent.com/14130041/142380903-d40596c8-0782-4432-a04b-12e9221bdb27.png)

This is from FreeOPCUA:
![image](https://user-images.githubusercontent.com/14130041/142383192-4a40c383-f711-45cc-ab15-2390b49715a2.png)

```
08:51:40.400Z :argument_list                 :136    checking argDefDataType  
 ...                                                          nodeId              : ns=0;i=852
 ...                                                          nodeClass           : DataType (64)
 ...                                                          browseName          : ServerState
 ...                                                          displayName         : null ServerState
 ...                                                          description         : locale=null text=
 ...                                                          isAbstract          : false
 ...                                                          definitionName      : 
 ...                                                          binaryEncodingNodeId: <none>
 ...                                                          xmlEncodingNodeId   : <none>
 ...                                                          subtypeOfObj        : Enumeration
 ...                                                          references    :   length =2
 ...                                                               +-> ------- HasProperty (ns=0;i=46) ------> [ns=0;i=7612             ]EnumStrings                              EnumStrings
 ...                                                               +-> <------- HasSubtype (ns=0;i=45) ------- [ns=0;i=29               ]Enumeration                              Enumeration
 ...                                                          back_references     :   length =0 ( references held by other nodes involving this node)
 ...                                                 Definition                   :
 ...                                                                              :   { /*EnumDefinition*/
 ...                                                                              :    fields                        /* EnumField        [] */: [
 ...                                                                              :      { /*0*/
 ...                                                                              :        value                     /* Int64               */: 0,0
 ...                                                                              :        displayName               /* LocalizedText       */: locale=null text=null
 ...                                                                              :        description               /* LocalizedText       */: locale=null text=null
 ...                                                                              :        name                      /* UAString            */: Running
 ...                                                                              :      },
 ...                                                                              :      { /*1*/
 ...                                                                              :        value                     /* Int64               */: 0,1
 ...                                                 .... TRUNCATED .....
08:51:40.410Z :argument_list                 :137    checking argDataType
 ...                                                          nodeId              : ns=0;i=6
 ...                                                          nodeClass           : DataType (64)
 ...                                                          browseName          : Int32
 ...                                                          displayName         : null Int32
 ...                                                          description         : locale=null text=
 ...                                                          isAbstract          : false
 ...                                                          definitionName      : 
 ...                                                          binaryEncodingNodeId: <none>
 ...                                                          xmlEncodingNodeId   : <none>
 ...                                                          subtypeOfObj        : Integer
 ...                                                          references    :   length =1
 ...                                                               +-> <------- HasSubtype (ns=0;i=45) ------- [ns=0;i=27               ]Integer                                  Integer
 ...                                                          back_references     :   length =0 ( references held by other nodes involving this node)
08:51:40.418Z :argument_list                 :154    MADI argDataType.nodeId.value is not equal to argDefDataType.nodeId.value
08:51:40.419Z :argument_list                 :155    MADI   argDataType.nodeId.value  6
08:51:40.421Z :argument_list                 :156    MADI   argDefDataType.nodeId.value  852
08:51:40.422Z :argument_list                 :181    MADI isArgumentValid returned false!
```
_Remark: "MADI debugs were added by me._

**Gist**

 
gist:

**Context**

- ( ) my request is related to node-opcua acting as an OPCUA CLIENT
- (X) my request is related to node-opcua acting as an OPCUA SERVER

- ( ) I have installed node-opcua from source ( using git clone)
- (X) I have installed node-opcua as a package ( using npm install )
- ( ) I am using an application that uses node-opcua

    - ( ) node-red
    - ( ) other : please specify

- Device: PC
- OS version: Windows X
    - ( ) Windows : version : 19042.1348
    - ( ) Linux : version : ______
    - ( ) MacOs : version : ______
    - ( ) Raspbian: version : ______
    - ( ) Other : specify : ______

- Description of the other OPCUA system I am trying to connect to:

    - Name: UAExpert
    - Version: 1.5.1
    - Manufacturer/Software vendor: unifiedautomation
    - link to the manufacturer or product site: http://www.unifiedautomation.com/

    - Name: FreeOPCUA
    - Version: 0.8.0 
![image](https://user-images.githubusercontent.com/14130041/142384569-30aa08e3-797a-4ef2-9256-718b3abbabc0.png)
    - Manufacturer/Software vendor: FreeOPCUA Library
    - link to the manufacturer or product site: http://freeopcua.github.io/

- **node-opcua version:** : 2.59.0 (tried with 2.18.0 and 2.40.0 as well)
  
  

- **Node:**
  `node --version` = 14.15.5

**Comment by erossignon** [comment]
What is your node-opcua version?

**Comment by marcodisarno** [comment]
latest - 2.59.0

**Comment by Graos** [comment]
same as Writing to an Enum Value from external OPC UA client leads to error in node-opcua server bug Sponsors welcomed
#849

**Comment by erossignon** [comment]
fixed in node-opcua@2.60.2
